support-frontend/assets/pages/paper-subscription-checkout/helpers/paymentProviders.js

```diff
@@ -30,7 +90,17 @@ function onPaymentAuthorised(paymentAuthorisation: PaymentAuthorisation, dispatc
   };
 
   dispatch(setFormSubmitted(true));
-  Promise.resolve({ paymentStatus: 'success' }).then(handleSubscribeResult);
+
+  const data = buildRegularPaymentRequest(state, paymentAuthorisation);
+
+  postRegularPaymentRequest(
+    routes.digitalSubscriptionCreate,
+    data,
+    state.common.abParticipations,
+    state.page.csrf,
+    () => {},
+    () => {},
```

Comment:
what are these two noops?

Reply:
They're both only used by the contributions checkout, something to do with setting a guest account token, and a thank-you page state. This is another candidate for a refactor, but probably not in the scope of this PR.

Reply:
This is another place where the original code is written with a few contribution specific assumptions which don't apply to subs. The first supports guest checkout and the second sets the thank you page stage. The second should probably be refactored to apply to all checkouts but not in this PR